k2/csrc/fsa_equivalent_test.cc

```diff
@@ -0,0 +1,61 @@
+// k2/csrc/fsa_equivalent_test.cc
+
+// Copyright (c)  2020  Haowen Qiu
+
+// See ../../LICENSE for clarification regarding multiple authors
+
+#include "k2/csrc/fsa_equivalent.h"
+
+#include <utility>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "k2/csrc/fsa.h"
+
+namespace k2 {
+
+TEST(Properties, RandomPathFail) {
+  {
+    Fsa fsa;
+    Fsa path;
+    bool status = RandomPath(fsa, &path);
+    EXPECT_FALSE(status);
+  }
+  // TODO(haowen): add tests for non-connected fsa
+}
+
+TEST(Properties, RandomPathSuccess) {
+  {
+    Fsa fsa;
+    std::vector<Arc> arcs = {
+        {0, 1, 1}, {0, 2, 2}, {1, 2, 3}, {2, 3, 4},
```

Comment:
To remove randomness, you can set up a case  such that every node has only one leaving arc,
so you can **check** the returned path.

Reply:
Sure, Add a case.